code/datums/components/payment.dm

```diff
@@ -0,0 +1,62 @@
+/**
+  * Handles simple payment operations where the cost of the object in question doesn't change.
+  *
+  * What this is useful for:
+  * Basic forms of vending.
+  * Objects that can drain the owner's money linearly.
+  * What this is not useful for:
+  * Things where the seller may want to fluxuate the price of the object.
+  * Improving standardizing every form of payment handing, as some custom handling is specific to that object.
+  **/
+/datum/component/payment
+	///Standardized of operation.
+	var/cost = 10
+	///Flavor style for handling cash (Friendly? Hostile? etc.)
+	var/transaction_style = "Clinical"
+	///Who's getting paid?
+	var/datum/bank_account/target
+
+/datum/component/payment/Initialize(_cost, _target, _style)
+	if(!_target)
+		target = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	target = _target
+	cost = _cost
+	transaction_style = _style
+	RegisterSignal(parent, list(COMSIG_OBJ_ATTEMPT_CHARGE_PRINTER, COMSIG_OBJ_ATTEMPT_CHARGE), .proc/attempt_charge)
+
+/datum/component/payment/proc/attempt_charge(datum/source, atom/movable/AM)
```

Comment:
The return type of a signal should be bitflags.

[You can see that boolean return values are unacceptable in this post here by Ninja.](https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674&p=524299#p494359)